src/Features/Core/Portable/FindUsages/ExternalReferenceItem.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.FindUsages
+{
+    /// <summary>
+    /// Information about a symbol's reference that can be used for display and navigation in an
+    /// editor.  These generally reference items outside of the Roslyn <see cref="Solution"/> model
+    /// provided by external sources such as online indices.
+    /// </summary>
+    internal abstract class ExternalReferenceItem
+    {
+        /// <summary>
+        /// The definition this reference corresponds to.
+        /// </summary>
+        public DefinitionItem Definition { get; }
+
+        public string ProjectName { get; }
+        public string DisplayPath { get; }
```

Comment:
Does this also need some of the other things in SourceReferenceItem? The usage info or AdditionalProperties?

Kinda wishing still this was extracted out where ReferenceItem still has a lot of the basics; ExternalReferenceItem and SourceReferenceItem still can live around but then just become derived types.

Reply:
> The usage info or AdditionalProperties?

Nope.  We don't have any of that info in richnav.

> Kinda wishing still this was extracted out where ReferenceItem still has a lot of the basics; ExternalReferenceItem and SourceReferenceItem still can live around but then just become derived types.

FindRefs would still have to explicitly cast the base type to these types to get at the data though.